### PR TITLE
Migrate the CI runners to the new clusters

### PR DESCRIPTION
**Pull request by glegendre01**
# What does this PR do?

Migrate the CI runners to the new clusters.

### CODE REVIEW

File: .github/workflows/self-pr-slow-ci.yml [hunk 1]
```diff
@@ -113,23 +129,23 @@ jobs:
...
-        run: cat /transformers/reports/${{ matrix.machine_type }}_run_models_gpu_${{ matrix.folders }}_test_reports/failures_short.txt
+        run: cat /transformers/reports/${{ env.machine_type }}_run_models_gpu_${{ matrix.folders }}_test_reports/failures_short.txt
 
       - name: Make sure report directory exists
         shell: bash
         run: |
-          mkdir -p /transformers/reports/${{ matrix.machine_type }}_run_models_gpu_${{ matrix.folders }}_test_reports
-          echo "hello" > /transformers/reports/${{ matrix.machine_type }}_run_models_gpu_${{ matrix.folders }}_test_reports/hello.txt
-          echo "${{ matrix.machine_type }}_run_models_gpu_${{ matrix.folders }}_test_reports"
+          mkdir -p /transformers/reports/${{ nv.machine_type }}_run_models_gpu_${{ matrix.folders }}_test_reports
```
Comment (ydshieh) [inline .github/workflows/self-pr-slow-ci.yml:142]:
```suggestion
          mkdir -p /transformers/reports/${{ env.machine_type }}_run_models_gpu_${{ matrix.folders }}_test_reports
```

File: .github/workflows/self-push.yml [hunk 2]
```diff
@@ -352,6 +390,23 @@ jobs:
           echo "env.CI_BRANCH = ${{ env.CI_BRANCH }}"
           echo "env.CI_SHA = ${{ env.CI_SHA }}"
 
+      - name: Set `machine_type` for report and artifact names
+        working-directory: /transformers
```
Comment (ydshieh) [inline .github/workflows/self-push.yml:394]:
```suggestion
        working-directory: /workspace/transformers
```

File: .github/workflows/self-push.yml [hunk 3]
```diff
@@ -444,6 +500,23 @@ jobs:
...
```
Comment (ydshieh) [inline .github/workflows/self-push.yml:504]:
```suggestion
        working-directory: /workspace/transformers
```

File: utils/tests_fetcher.py [hunk 4]
```diff
@@ -1153,6 +1153,8 @@ def parse_commit_message(commit_message: str) -> Dict[str, bool]:
 
 
 def create_test_list_from_filter(full_test_list, out_path):
+    if not os.path.exists(out_path):
+        os.makedirs(out_path)
```
Comment (ydshieh) [inline utils/tests_fetcher.py:1157]:
```suggestion
    os.makedirs(out_path, exist_ok=True)
```

**Review by ydshieh — APPROVED** [review]
Thanks a lot of this update! LGTM except a few nits.

File: .github/workflows/self-push.yml (hunk 2 above)
Comment (ydshieh) [reply .github/workflows/self-push.yml:394]: this image (deepspeed) is somehow a bit different

**Review by LysandreJik — APPROVED** [review]
Thank you!

**Pull request merged**